### PR TITLE
Add Evidence and Discussion reference types

### LINKED ISSUE #78: Additional options for references field to support reproducers/exploit code
This is in relation to: https://github.com/cloudsecurityalliance/gsd-database/issues/2389

Ok meta comments:

Obviously, we want this added as a reference, but the OSV schema only supports:

https://ossf.github.io/osv-schema/#references-field

ADVISORY: A published security advisory for the vulnerability.
ARTICLE: An article or blog post describing the vulnerability.
REPORT: A report, typically on a bug or issue tracker, of the vulnerability.
FIX: A source code browser link to the fix (e.g., a GitHub commit) Note that the fix type is meant for viewing by people using web browsers. Programs interested in analyzing the exact commit range would do better to use the GIT-typed affected[].ranges entries (described above).
PACKAGE: A home web page for the package.
EVIDENCE: A demonstration of the validity of a vulnerability claim, e.g. app.any.run replaying the exploitation of the vulnerability.
WEB: A web page of some unspecified kind.

So it could be classed as "EVIDENCE", which is what I'll use for now, but I'm going to file an upstream issue with OSV to try for a better word, e.g.:

REPRODUCER: Nonweaponized, programmatic method to trigger the vulnerability?
EXPLOIT: weaponized, programmatic method to trigger the vulnerability (can be a link to code?)
EXPLOITATION: reports of exploitation but no actual technical reproducer/exploit available?

I think one possible way to do this would be to use "sub tags" e.g. EVIDENCE:EXPLOIT or EVIDENCE:REPRODUCER. This would prevent clutter and an explosion of top level tags, and for data consumers they can for example simply support "EVIDENCE" and ignore subtags, or choose to also support subtags, either way they are able to broadly classify the data according to the top level tags, even if they don't want to check for subtags.

### PR DESCRIPTION
**Pull request by joshbuker**
Also normalized the ordering for the types to be alphabetical order.

Closes #78

### CODE REVIEW

File: docs/schema.md [hunk 1]
```diff
@@ -690,14 +690,17 @@ The known reference `type` values are:
 
 - `ADVISORY`: A published security advisory for the vulnerability.
 - `ARTICLE`: An article or blog post describing the vulnerability.
-- `REPORT`: A report, typically on a bug or issue tracker, of the vulnerability.
+- `DETECTION`: An article or code that provides detection of the vulnerability.
```
Comment (oliverchang) [inline docs/schema.md:693]: What does "article or code that provides detection" mean exactly? 

It seems a bit ambiguous to me -- does it mean:

- "Code" that reproduces the issue (in which case `EVIDENCE` seems like a better fit?), or 
- An article that talks about how this vuln was discovered?

**Review by oliverchang — COMMENTED** [review]
Thanks! Apologies again for the delay -- I was out on vacation for a few weeks.

File: docs/schema.md (hunk 1 above)
Comment (chrisbloom7) [reply docs/schema.md:693]: I agree. It sounds like this could already be represented by either `ARTICLE` or `EVIDENCE`.

Comment (joshbuker) [reply docs/schema.md:693]: @kurtseifried Do you want to expand on this?

Comment (kurtseifried) [reply docs/schema.md:693]: so for example YARA rules, semgrep, etc. e.g. https://gsd.id/GSD-2021-44228 

est detection:

grep: https://gist.github.com/Neo23x0/e4c8b03ff8cdf1fa63b7d15db6e3860b

jarhashes: https://github.com/mubix/CVE-2021-44228-Log4Shell-Hashes

semgrep: https://github.com/returntocorp/semgrep-rules/pull/1650/commits/ecfc32623eec718d61ec83b9196574f333191008/

yara: https://github.com/timb-machine/log4j/

burpsuite: https://github.com/silentsignal/burp-log4shell

Nmap NSE: https://github.com/Diverto/nse-log4shell

**Comment by joshbuker** [comment]
https://github.com/cloudsecurityalliance/gsd-tools/pull/197

**Pull request closed without merging**